addons/mail/static/src/owl/components/activity/activity.xml

```diff
@@ -0,0 +1,147 @@
+                                            t-on-click="_onClickPreviewMailTemplate">
+                                        Preview
+                                    </button>
+                                    <span class="o_Activity_mailTemplate_text">or</span>
+                                    <button class="o_Activity_mailTemplate_button o_Activity_mailTemplate_send btn btn-link"
+                                            t-att-data-mail-template-id="mailTemplate.id"
+                                            t-on-click="_onClickSendMailTemplate">
+                                        Send Now
+                                    </button>
+                                </div>
+                            </t>
+                        </div>
+                    </t>
+
+                    <t t-if="storeProps.activity.canWrite">
+                        <div class="o_Activity_tools">
+                            <t t-if="storeProps.activity.activityCategory !== 'upload_file'">
+                                <ActivityMarkDoneButton activityLocalId="props.activityLocalId" class="o_Activity_toolButton o_Activity_markDoneButton"/>
+                            </t>
+                            <t t-else="">
+                                <button class="o_Activity_toolButton o_Activity_uploadButton btn btn-link"
+                                        t-on-click="_onClickUploadDocument">
+                                    <i class="fa fa-upload"/>
+                                    Upload Document
+                                </button>
+                                <FileUploader
+                                    attachmentLocalIds="storeProps.activity.attachmentLocalIds"
+                                    uploadId="storeProps.activity.resId"
+                                    uploadModel="storeProps.activity.model"
+                                    t-on-o-attachment-created="_onAttachmentCreated"
+                                    t-ref="fileUploader"
+                                />
+                            </t>
+                            <button class="o_Activity_toolButton o_Activity_editButton btn btn-link"
+                                    t-on-click="_onClickEditActivity">
+                                <i class="fa fa-pencil"/>
+                                Edit
+                            </button>
+                            <button class="o_Activity_toolButton o_Activity_cancelButton btn btn-link"
+                                    t-on-click="_onClickCancelActivity">
+                                <i class="fa fa-times"/>
+                                Cancel
+                            </button>
+                        </div>
+                    </t>
+                </div>
+            </div>
+        </t>
+        <!-- Need to print something so that render is happy if an empty activity is rendered -->
+        <t t-else="">
+            <div />
+        </t>
```

Comment:
This shouldn't happen... Activity component make only sense when it is related to an activity!